src/Form/Form.js

```diff
@@ -410,10 +449,14 @@ class Form {
 
         // If the form has too many inputs, destroy everything to avoid performance issues
         if (this.inputs.all.size > MAX_INPUTS_PER_FORM) {
-            if (shouldLog()) {
-                console.log('The form has too many inputs, destroying.')
-            }
-            this.destroy()
+            this.device.scanner.stopScanner('The form has too many inputs, destroying.')
+            return this
+        }
+
+        // When new inputs are added after the initial scan, reanalyze the whole form
+        if (this.initialScanComplete && this.rescanCount < MAX_FORM_RESCANS) {
```

Comment:
Checking the counter here as well, to avoid running the FormAnalyzer again, which is very expensive when the form doesn't have good markup.